server/status.go

```diff
@@ -188,16 +192,58 @@ func (s *statusServer) handleNodesStatus(w http.ResponseWriter, r *http.Request)
 	w.Write(b)
 }
 
-// handleNodeStatus handles GET requests for a single node status.
+// handleNodeStatus handles GET requests for a single node's status. If no id is
+// available, it calls handleNodesStatus to return all node's statuses.
 func (s *statusServer) handleNodeStatus(w http.ResponseWriter, r *http.Request) {
-	// TODO(bram) parse node-id in path and return the single node status only.
+	// TODO(bram) parse node-id in path and return the single node's status
+	// only.
 	s.handleNodesStatus(w, r)
 }
 
-// handleStoresStatus handles GET requests for store status.
+// handleStoresStatus handles GET requests for all store's statuses.
 func (s *statusServer) handleStoresStatus(w http.ResponseWriter, r *http.Request) {
-	w.Header().Set("Content-Type", "application/json")
-	w.Write([]byte(`{"stores": []}`))
+	startKey := engine.KeyStatusStorePrefix
+	endKey := startKey.PrefixEnd()
+
+	call := client.Scan(startKey, endKey, 0)
```

Comment:
Do you think it's worthwhile to have a single handler helper function which handles nodes and stores but accepts the applicable key prefix? These two sorts of status are probably going to remain quite equivalent.